src/client.cc

```diff
@@ -23,6 +23,13 @@
 namespace Kakoune
 {
 
+static Buffer::FsStatus get_fs_status(StringView filename)
```

Comment:
I implemented this function here, instead of `file.cc` because of the `Buffer::FsStatus` type (probably don't want that visible in the generic file-related API).

Reply:
Well, FsStatus does not really need to be buffer specific, it could make sense to define it in file.hh along with `get_fs_status` and just use it in the buffer.